cf_xarray/accessor.py

```diff
@@ -1340,23 +1340,36 @@ def rename_like(
         theirkeys = other.cf.keys()
 
         good_keys = ourkeys & theirkeys
-        renamer = {}
+        keydict = {}
         for key in good_keys:
             ours = _single(_get_all)(self._obj, key)[0]
             theirs = _single(_get_all)(other, key)[0]
-            renamer[ours] = theirs
-
-        conflicts = {
-            k: v for k, v in renamer.items() if list(renamer.values()).count(v) > 1
-        }
+            keydict[key] = dict(ours=ours, theirs=theirs)
+
+        conflicts = {}
+        for k0, v0 in keydict.items():
+            for v1 in keydict.values():
+                # Conflicts have same ours but different theirs or vice versa
+                if sum([v0["ours"] == v1["ours"], v0["theirs"] == v1["theirs"]]) == 1:
```

Comment:
This is actually also addressing this: https://github.com/xarray-contrib/cf-xarray/pull/183/files/2a7d312dfa877a624ecdd837029559901fde9cc7#r583839911